torch/csrc/jit/script/compiler.cpp

```diff
@@ -636,35 +638,42 @@ at::optional<MatchedSchema> tryMatchSchema(
     return MatchedSchema {std::move(positional_inputs), std::move(return_types) };
 }
 
-
-static Value* tryEmitBuiltin(
-  const std::shared_ptr<Operator>& op,
-  std::stringstream& failure_messages,
-  const SourceRange& loc,
-  Graph& graph,
-  Symbol name,
-  at::ArrayRef<NamedValue> inputs,
-  at::ArrayRef<NamedValue> attributes,
-  bool convert_tensors_to_nums) {
-
-  auto matched_schema = tryMatchSchema(op->schema(), loc, graph, inputs, attributes,
-    failure_messages, convert_tensors_to_nums);
-  if(!matched_schema)
-    return nullptr;
-  // we successfully matched this schema, construct the node
-
-  auto n = graph.insertNode(graph.create(name, matched_schema->inputs, 0))
-                ->setSourceLocation(std::make_shared<SourceRange>(loc));
-
-  for(auto & ret : matched_schema->return_types) {
-    n->addOutput()->setType(ret);
-  }
-
-  // assert that we did indeed create an op that has implementation
-  // otherwise schema and dispatch are not in sync
-  getOperation(n);
-
-  return packOutputs(graph, n->outputs());
+// Match against a potentially mutable schema.
+//
+// We need to treat mutable schemas differently because the IR explicitly
+// expresses effects by including a world token in mutable ops. Users do not
+// know about the world token, so we need to generate a dummy one and add
+// it to the inputs for schema matching.
+//
+// Example:
+//   append(int[] list, int el)
+// becomes
+//   append(World w, int[] list, int el)
+//
+// NOTE: The dummy world token has no meaning; the AnnotateEffects pass is
+// necessary to enforce linearization on effectful ops.
+at::optional<MatchedSchema> tryMatchSchema(
+    const FunctionSchema& schema,
+    const SourceRange& loc,
+    Graph& graph,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    std::ostream& failure_messages,
+    bool convert_tensors_to_nums) {
+  std::vector<NamedValue> modifiedInputs(inputs.begin(), inputs.end());
```

Comment:
Can you just put these lines at the beginning of tryMatchSchema. There are like 20 lines of parameter passing to the Impl version to add these 5 lines of code here.

Reply:
oops, this function used to be more complicated I think. Will remove